frontend/public/components/secrets/create-secret/WebHookSecretForm.tsx

```diff
@@ -1,86 +1,75 @@
-  }
-  changeWebHookSecretkey(event) {
-    this.setState(
-      {
-        stringData: { WebHookSecretKey: event.target.value },
-      },
-      () => this.props.onChange(this.state),
-    );
-  }
-  generateWebHookSecret() {
-    this.setState(
-      {
-        stringData: { WebHookSecretKey: generateSecret() },
-      },
-      () => this.props.onChange(this.state),
-    );
-  }
-  render() {
-    const { t } = this.props;
-    return (
-      <div className="form-group">
-        <label className="control-label co-required" htmlFor="webhook-secret-key">
-          {t('public~Webhook secret key')}
-        </label>
-        <div className="pf-v5-c-input-group">
-          <input
-            className="pf-v5-c-form-control"
-            id="webhook-secret-key"
-            data-test="secret-key"
-            type="text"
-            name="webhookSecretKey"
-            onChange={this.changeWebHookSecretkey}
-            value={this.state.stringData.WebHookSecretKey}
-            aria-describedby="webhook-secret-help"
-            required
-          />
-          <button
-            type="button"
-            onClick={this.generateWebHookSecret}
-            className="pf-v5-c-button pf-m-tertiary"
-            data-test="webhook-generate-button"
-          >
-            {t('public~Generate')}
-          </button>
-        </div>
-        <p className="help-block" id="webhook-secret-help">
-          {t('public~Value of the secret will be supplied when invoking the webhook.')}
-        </p>
-      </div>
-    );
-  }
-}
+export const WebHookSecretForm: React.FC<SecretSubFormProps> = ({ onChange, stringData }) => {
+  const { t } = useTranslation();
 
-export const WebHookSecretForm = withTranslation()(WebHookSecretFormWithTranslation);
+  const handleWebHookSecretChange = (newSecret: string) => {
+    onChange({ stringData: { ...stringData, WebHookSecretKey: newSecret }, base64StringData: {} });
+  };
 
-type WebHookSecretFormState = {
-  stringData: {
-    [key: string]: string;
+  const changeWebHookSecretkey = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newSecret = event.target.value;
+    handleWebHookSecretChange(newSecret);
   };
-};
 
-type WebHookSecretFormProps = {
-  onChange: Function;
-  onError: Function;
-  onFormDisable: Function;
-  secretType: string;
-  stringData: {
-    [key: string]: string;
+  const generateWebHookSecret = () => {
+    // the current (4.5.5) version of typescript does not have the randomUUID function defined, thus it produces ts errors
+    // but all of our supported browsers should have it defined, so we can safely ignore the error.
+    // randomUUID is firstly defined in 4.6 version of typescript, so once we upgrade to this or newer version we can remove this ts-ignore
+    // @ts-ignore-next-line
+    const newSecret = window.crypto.randomUUID();
+    handleWebHookSecretChange(newSecret);
   };
+
+  return (
```

Comment:
```suggestion
  return (
    <FormGroup fieldId="webhook-secret-key" isRequired label={t('public~Webhook secret key')}>
      <InputGroup>
        <InputGroupItem isFill>
          <TextInput
            id="webhook-secret-key-input"
            data-test="secret-key"
            type="text"
            name="webhookSecretKey"
            onChange={changeWebHookSecretkey}
            value={stringData.WebHookSecretKey}
            aria-describedby="webhook-secret-help"
            isRequired
          />
        </InputGroupItem>
        <InputGroupItem>
          <Button
            id="webhook-secret-key-generate-button"
            onClick={generateWebHookSecret}
            data-test="webhook-generate-button"
            variant="tertiary"
          >
            {t('public~Generate')}
          </Button>
        </InputGroupItem>
      </InputGroup>
      <FormHelperText id="webhook-secret-help">
        <HelperText>
          <HelperTextItem>
            {t('public~Value of the secret will be supplied when invoking the webhook.')}
          </HelperTextItem>
        </HelperText>
      </FormHelperText>
    </FormGroup>
  );
```

I think it might be better to use the `InputGroup` PF component instead of `Grid`s. There is also another issue I observed where now this section of the form looks inconsistent with the rest of the form, but I think that could be addressed in another ticket lol

Reply:
Right call, this is exact use case for `InputGroup`, didn't know about that one. Yeah the inconsistency is expected, the other ("Secret name") section is in the parent component that wraps all of the Secret subforms, and as you said its gonna be addressed in [another ticket ](https://issues.redhat.com/browse/CONSOLE-4082).